### PR TITLE
Automate velero PR

### PR DESCRIPTION
**Pull request by areed**
[Example PR](https://github.com/replicatedhq/kURL/pull/1277)

### CODE REVIEW

**Review by DanStough — APPROVED** [review]

File: addons/velero/template/base/install.sh [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        return 0
+    fi
+
+    if [ "${K8S_DISTRO}" != "rke2" ]; then
+        return 0
+    fi
+
+    render_yaml_file "$src/restic-daemonset-privileged.yaml" > "$dst/restic-daemonset-privileged.yaml"
+    inserpt_patches_strategic_merge "$dst/kustomization.yaml" restic-daemonset-privileged.yaml
+}
```
Comment (DanStough) [inline addons/velero/template/base/install.sh:97]: Do we need `VELERO_RESTIC_REQUIRES_PRIVILEGED` anymore if this is how we manage it? I like that you don't need to remember to set it.

Comment (DanStough) [reply addons/velero/template/base/install.sh:97]: If we decide to take it out (now or at a later point) need to update the RKE2 Testgrid run.

Comment (areed) [reply addons/velero/template/base/install.sh:97]: I didn't realize that was part of the kurl spec. I'll restore it.

File: addons/velero/template/base/install.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    $src/assets/velero-v${VELERO_VERSION}-linux-amd64/velero install --crds-only | grep -v 'Velero is installed'
+
+    local resticArg="--use-restic"
+    if [ "$VELERO_DISABLE_RESTIC" = "1" ]; then
+        resticArg=""
+    fi
+
+    local bslArgs="--no-default-backup-location"
+    if ! kubernetes_resource_exists "$VELERO_NAMESPACE" backupstoragelocation default; then
+        bslArgs="--provider aws --bucket $VELERO_LOCAL_BUCKET --backup-location-config region=us-east-1,s3Url=${OBJECT_STORE_CLUSTER_HOST},publicUrl=http://${OBJECT_STORE_CLUSTER_IP},s3ForcePathStyle=true"
```
Comment (salahalsaleh) [inline addons/velero/template/base/install.sh:46]: `s3ForcePathStyle=true` most probably needs quotes around `true`

File: addons/velero/template/base/install.sh [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    fi
+
+    velero_credentials
+
+    $src/assets/velero-v${VELERO_VERSION}-linux-amd64/velero install \
+        $resticArg \
+        $bslArgs \
+        --plugins velero/velero-plugin-for-aws:v1.1.0,velero/velero-plugin-for-gcp:v1.1.0,velero/velero-plugin-for-microsoft-azure:v1.1.0 \
+        --secret-file velero-credentials \
+        --use-volume-snapshots=false \
```
Comment (salahalsaleh) [inline addons/velero/template/base/install.sh:56]: we use `restic` to backup volume data, so i believe this needs to use the default `true` value

File: addons/velero/template/base/install.sh [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    fi
+
+    local bslArgs="--no-default-backup-location"
+    if ! kubernetes_resource_exists "$VELERO_NAMESPACE" backupstoragelocation default; then
+        bslArgs="--provider aws --bucket $VELERO_LOCAL_BUCKET --backup-location-config region=us-east-1,s3Url=${OBJECT_STORE_CLUSTER_HOST},publicUrl=http://${OBJECT_STORE_CLUSTER_IP},s3ForcePathStyle=true"
+    fi
+
+    velero_credentials
+
+    $src/assets/velero-v${VELERO_VERSION}-linux-amd64/velero install \
```
Comment (salahalsaleh) [inline addons/velero/template/base/install.sh:51]: this install command is missing the `--use-restic` flag

File: addons/velero/template/base/install.sh (hunk 3 above)
Comment (areed) [reply addons/velero/template/base/install.sh:56]: Volumes snapshots usually integrate with a cloud provider, e.g. EBS on AWS. We don't have a VolumeSnapshotLocation in earlier versions of velero on kurl.

Comment (salahalsaleh) [reply addons/velero/template/base/install.sh:56]: ah, i see.

**Review by salahalsaleh — APPROVED** [review]

File: addons/velero/template/base/install.sh (hunk 2 above)
Comment (areed) [reply addons/velero/template/base/install.sh:46]: It seems to end up quoted in kubectl
```
spec:
  config:
    publicUrl: http://10.96.1.229
    region: us-east-1
    s3ForcePathStyle: "true"
    s3Url: http://rook-ceph-rgw-rook-ceph-store.rook-ceph
  objectStorage:
    bucket: velero
  provider: aws
```

Comment (salahalsaleh) [reply addons/velero/template/base/install.sh:46]: nice, just wanted to confirm, thanks!

**Pull request merged**